splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsOperation.java

```diff
@@ -126,6 +126,9 @@ else if (storedAs.equals("O"))
             } else {
                 statsDataSet = scanSetBuilder.buildDataSet(scope).map(new CloneFunction<>(operationContext));
             }
+            if(statsDataSet.partitions() == 0) {
```

Comment:
do we test this case?

Reply:
Yes, ExternalTableIT.collectStats() was failing due to this since DB-9369 was merged